### PR TITLE
[KB-288] Experiment design for initial experiment

### PR DESCRIPTION
**Pull request by Ivikhostrup**
Reading guide:
Its a rather short section, but I felt like there was not too much to say other than what we wanted to do and just coming up with the arguments for these choices.

Ive added to the experiment section that all runs are done using our hyperparameter optimization tool.

### CODE REVIEW

File: report_thesis/src/sections/appendix.tex [hunk 1]
```diff
@@ -43,4 +43,251 @@ \subsection{Cross-Validation Plots for Major Oxides}\label{subsec:cv_plots}
...
+\caption{Hyperparameters for NGBoost}
+\begin{tabular}{ll}
+    \toprule
+    \textbf{Hyperparameter} & \textbf{Value} \\
+    \midrule
+    n/a & n/a \\
+    \bottomrule
+\end{tabular}
+\label{tab:ngboost_hyperparameters}
+\end{table}
```
Comment (chhoumann) [inline report_thesis/src/sections/appendix.tex:237]: ?

image

File: report_thesis/src/sections/appendix.tex [hunk 2]
```diff
@@ -43,4 +43,251 @@ \subsection{Cross-Validation Plots for Major Oxides}\label{subsec:cv_plots}
...
+    Layer 12: MaxPooling1D & (pool size: 2) \\
+    Layer 13: Conv1D & (filters: 128, kernel size: 3, activation: relu) \\
+    Layer 14: Add & (merge with layer 12 output) \\
+    Layer 15: Flatten & (output) \\
+    Layer 16: Dropout & (rate: 0.5) \\
+    Layer 17: Dense & (512 units, activation: relu) \\
+    Layer 18: Dense & (1 unit, activation: linear) \\
+    Total Parameters & 25,274,245 \\
+    Trainable Parameters & 25,274,243 \\
+    Non-trainable Parameters & 2 \\
```
Comment (chhoumann) [inline report_thesis/src/sections/appendix.tex:288]: Probably not relevant?

File: report_thesis/src/sections/experiment_design/index.tex [hunk 3]
```diff
@@ -0,0 +1,8 @@
+\section{Experimental Design}\label{sec:methodology}
+In this section, we outline the methodology used in this study to address the challenges identified in Section~\ref{sec:problem_definition}. Our objective was to identify the most promising machine learning models and preprocessing techniques proposed from the literature, as outlined in Section~\ref{sec:related-work}, for predicting major oxide compositions from \gls{libs} data.
+Then, using this knowledge, develop a pipeline that utilizes the strengths of these models and preprocessing techniques to improve prediction accuracy and robustness of the predictions.
+We first describe the datasets used, including their preparation and the method of splitting for model training. Next, we outline the preprocessing steps and the model selection process, followed by a detailed explanation of the experimental setup and evaluation metrics. Finally, we discuss our validation testing procedures and the approach taken to ensure unbiased final model evaluations.
```
Comment (chhoumann) [inline report_thesis/src/sections/experiment_design/index.tex:4]: This introduction is really weird. It feels out of place & like it describes an entirely different section.
Need to rewrite.

I think this is a leftover from before we made a bunch of changes?

File: report_thesis/src/sections/experiment_design/initial_experiment.tex [hunk 4]
```diff
@@ -0,0 +1,18 @@
+\subsection{Design for Initial Experiment}\label{sec:initial-experiment}
+The goal of the initial experiment is to identify the most promising models for further evaluation.
```
Comment (chhoumann) [inline report_thesis/src/sections/experiment_design/initial_experiment.tex:2]: Past tense.

```suggestion
The goal of the initial experiment was to identify the most promising models for further evaluation.
```

File: report_thesis/src/sections/experiment_design/initial_experiment.tex [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+This experiment was designed to give an initial indication of the performance of the models on the \gls{ccs} dataset.
+
+To conduct the initial experiment and evaluate the twelve selected models shown in Table~\ref{tab:preprocessing-models} we wanted to ensure that the models were evaluated on a consistent basis.
```
Comment (chhoumann) [inline report_thesis/src/sections/experiment_design/initial_experiment.tex:5]: Past tense.
```suggestion
To conduct the initial experiment and evaluate the twelve selected models shown in Table~\ref{tab:preprocessing-models}, we ensured that the models were evaluated on a consistent basis.
```

File: report_thesis/src/sections/experiment_design/initial_experiment.tex [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+We achieve this by setting the following conditions for the experiment.
+The only preprocessor used for all models was \texttt{Norm 3} normalization.
+This normalization technique was developed specifically by \citet{cleggRecalibrationMarsScience2017} for accomodating the characteristics of the \gls{libs} data, whereby the data represents the readings of three different spectrometers, as outlined in Section~\ref{sec:norm3}.
+Therefore, \texttt{Norm 3} represents the most \gls{libs}-specific normalization technique available in from the pool of selected preprocessors and is the most appropriate choice for this experiment.
```
Comment (chhoumann) [inline report_thesis/src/sections/experiment_design/initial_experiment.tex:9]: Past tense + it isn't a characteristic of LIBS data, but the collection process. 
They have 5 spectrometers on SuperCam, and use Norm 5 there. It's a general approach.

```suggestion
This normalization technique was developed by \citet{cleggRecalibrationMarsScience2017} to accommodate the characteristics of how the \gls{libs} data is collected, whereby the data represented the readings of three different spectrometers, as outlined in Section~\ref{sec:norm3}.
Therefore, \texttt{Norm 3} represented the most \gls{libs}-specific normalization technique available from the pool of selected preprocessors and was the most appropriate choice for this experiment.
```

Comment (chhoumann) [reply report_thesis/src/sections/experiment_design/initial_experiment.tex:9]: > Therefore, \texttt{Norm 3} represented the most \gls{libs}-specific normalization technique available from the pool of selected preprocessors and was the most appropriate choice for this experiment.

I'm not sure how I feel about this argument. Not great, but I don't really have anything much better. Can we discuss?

File: report_thesis/src/sections/experiment_design/initial_experiment.tex [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+All models were trained using our data partitioning and cross-validation strategy, as described in Section~\ref{subsec:validation_testing_procedures}. 
+To ensure as fair of a comparison between models as possible, all models were trained using as many default hyperparameters as possible and those hyperparameters that did not have default options were selected based on values found in the literature.
+However, due to the nature of the neural network models architecture, some extra time was spent on tuning the models to ensure a fair comparison.
```
Comment (chhoumann) [inline report_thesis/src/sections/experiment_design/initial_experiment.tex:12]: Grammar.

```suggestion
To ensure as fair of a comparison between models as possible, all models were trained using as many default hyperparameters as possible, and those hyperparameters that did not have default options were selected based on values found in the literature.
However, due to the nature of the neural network models' architecture, some extra time was spent on tuning the models to ensure a fair comparison.
```

File: report_thesis/src/sections/experiment_design/initial_experiment.tex [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+We achieve this by setting the following conditions for the experiment.
+The only preprocessor used for all models was \texttt{Norm 3} normalization.
+This normalization technique was developed specifically by \citet{cleggRecalibrationMarsScience2017} for accomodating the characteristics of the \gls{libs} data, whereby the data represents the readings of three different spectrometers, as outlined in Section~\ref{sec:norm3}.
+Therefore, \texttt{Norm 3} represents the most \gls{libs}-specific normalization technique available in from the pool of selected preprocessors and is the most appropriate choice for this experiment.
+All models were trained using our data partitioning and cross-validation strategy, as described in Section~\ref{subsec:validation_testing_procedures}. 
+To ensure as fair of a comparison between models as possible, all models were trained using as many default hyperparameters as possible and those hyperparameters that did not have default options were selected based on values found in the literature.
```
Comment (chhoumann) [inline report_thesis/src/sections/experiment_design/initial_experiment.tex:11]: > as many default hyperparameters as possible and those hyperparameters that did not have default options were selected based on values found in the literature.

Can we discuss this?

File: report_thesis/src/sections/experiment_design/initial_experiment.tex [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+However, due to the nature of the neural network models architecture, some extra time was spent on tuning the models to ensure a fair comparison.
+Finally, all the models ran only one trial per oxide.
```
Comment (chhoumann) [inline report_thesis/src/sections/experiment_design/initial_experiment.tex:13]: Don't like calling it trials here. That's a term we use for the optimization & it doesn't make sense in this context IMO.

We evaluated each model once per oxide given the selected configuration of hyperparameters.

File: report_thesis/src/sections/experiment_design/initial_experiment.tex [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+Therefore, \texttt{Norm 3} represents the most \gls{libs}-specific normalization technique available in from the pool of selected preprocessors and is the most appropriate choice for this experiment.
+All models were trained using our data partitioning and cross-validation strategy, as described in Section~\ref{subsec:validation_testing_procedures}. 
+To ensure as fair of a comparison between models as possible, all models were trained using as many default hyperparameters as possible and those hyperparameters that did not have default options were selected based on values found in the literature.
+However, due to the nature of the neural network models architecture, some extra time was spent on tuning the models to ensure a fair comparison.
+Finally, all the models ran only one trial per oxide. 
+Although running multiple trials would allow for an average performance evaluation, we argue that the variability of the results in each trial would be small enough that an average performance would not change the ranking of the models. 
+Additionally, as stated, the goal of this experiment is merely to get an initial indication of the performance of the models.
```
Comment (chhoumann) [inline report_thesis/src/sections/experiment_design/initial_experiment.tex:15]: Past tense.
```suggestion
Although running multiple trials would have allowed for an average performance evaluation, we argued that the variability of the results in each trial would be small enough that an average performance would not change the ranking of the models. 
Additionally, as stated, the goal of this experiment was merely to get an initial indication of the performance of the models.
```

Comment (chhoumann) [inline report_thesis/src/sections/experiment_design/initial_experiment.tex:15]: > Additionally, as stated, the goal of this experiment is merely to get an initial indication of the performance of the models.

This is enough as-is.
 I don't think we have to clarify that we don't run experiments multiple times. That's usually something you do for testing system performance. We're already doing cross-validation and have a separate test set.

**Review by chhoumann — COMMENTED** [review]
Agree with the general direction

File: report_thesis/src/sections/experiment_design/initial_experiment.tex (hunk 4 above)
Comment (Pattrigue) [reply report_thesis/src/sections/experiment_design/initial_experiment.tex]: Should addressed now that the intro was changed completely.

File: report_thesis/src/sections/experiment_design/initial_experiment.tex (hunk 6 above)
Comment (Pattrigue) [reply report_thesis/src/sections/experiment_design/initial_experiment.tex]: Same here, the intro was changed.

File: report_thesis/src/sections/experiment_design/experimental_setup.tex [hunk 11]
```diff
@@ -0,0 +1,7 @@
+\subsection{Experimental Setup}\label{sec:experimental_setup}
+Experiments were conducted on a machine equipped with an Intel Xeon Gold 6242 CPU, featuring 16 cores and 32 threads.
+The CPU has a base clock speed of 2.80 GHz and a maximum turbo frequency of 3.90 GHz.
+The system has 64 GB of RAM and runs on Ubuntu 22.04.2 LTS.
+Models were implemented using Python 3.10.11.
+The primary libraries used were Scikit-learn 1.4.2, XGBoost 2.0.3, Torch 2.2.2, NumPy 1.26.4, Pandas 2.2.1, Keras 3.2.1 and Optuna 3.6.1.
+Additionally, all experiments were run using the hyperparameter optimization tool described in Section~\ref{subsec:hyperparameter_tuning_tool}. % TODO: Add correct ref once other PR is in
```
Comment (Ivikhostrup) [inline report_thesis/src/sections/experiment_design/experimental_setup.tex:7]:
```suggestion
Additionally, all experiments were run using the hyperparameter optimization tool described in Section~\ref{sec:optimization_framework}.
```

**Review by chhoumann — APPROVED** [review]
